app/src/main/java/com/github/h3lp3rs/h3lp/HelpPageActivity.kt

```diff
@@ -125,15 +131,44 @@ class HelpPageActivity : AppCompatActivity(), CoroutineScope by MainScope() {
         }
     }
 
-
     /** Starts the activity by sending intent */
     private fun goToActivity(ActivityName: Class<*>?) {
         val intent = Intent(this, ActivityName)
         startActivity(intent)
     }
 
-    fun goToMainPage(view: View) {
-        goToActivity(MainPageActivity::class.java)
+    /**
+     * Triggered when clicked on the cross
+     */
+    fun refuseHelp(view: View) {
+        goToMainPage()
     }
 
+    /**
+     * Triggered when the helper accepts to help
+     * @param view The view of the button
+     */
+    fun acceptHelp(view: View) {
+        if(helpId == null) {
+            goToMainPage()
+            return
+        }
+        databaseOf(EMERGENCIES).getObject(helpId!!, EmergencyInformation::class.java).thenApply {
+            // Add the helper to the list of helpers
+            val me = Helper(userUid!!, currentLat, currentLong)
+            val helpers = ArrayList<Helper>()
+            if(!helpers.contains(me)) {
```

Comment:
I don't see the need for this check

Reply:
Whoosh, major bug here!
This is amazing since it is not catched my test because the reason is another bug in the existing code base...
Some exciting stuff :-)

Reply:
It was meant to be:
val helpers = ArrayList<Helper>(it.helpers)